junit5-engine/src/main/java/org/junit/gen5/engine/junit5/descriptor/JUnit5TestDescriptor.java

```diff
@@ -55,12 +56,12 @@ protected JUnit5TestDescriptor(UniqueId uniqueId) {
 		// @formatter:on
 	}
 
-	protected String determineDisplayName(AnnotatedElement element, String defaultName) {
+	protected String determineDisplayName(AnnotatedElement element, Supplier<String> defaultNameGenerator) {
 		// @formatter:off
 		return findAnnotation(element, DisplayName.class)
 				.map(DisplayName::value)
 				.filter(StringUtils::isNotBlank)
-				.orElse(defaultName);
+				.orElse(defaultNameGenerator.get());
```

Comment:
Ummmm.... yeah.... except that the argument is now `Function<AnnotatedElement, String>`. 😉